src/vs/base/browser/ui/tree/abstractTree.ts

```diff
@@ -1416,6 +1416,79 @@ export abstract class AbstractTree<T, TFilterData, TRef> implements IDisposable
 		this.model.setCollapsed(this.model.rootRef, true, true);
 	}
 
+	collapseFromLevel(level: number): void {
```

Comment:
All these methods look like utils. I believe you can implement them outside of a tree, by using the tree's API so I would recommend doing that.

Reply:
To what file/class should they go?